dts/bindings/sensor/st,stm32-qdec.yaml

```diff
@@ -70,3 +70,13 @@ properties:
     description: |
       This is a number >= 1 that is used to determine how many revolutions
       were done based on the current counter's value.
+
+  st,encoder-mode:
+    type: int
+    description: Encoder counting mode
+    default: 0
+    enum:
+      - 0 # MODE_X2_TI1 Counter counts up/down on TI2FP2 edge depending on TI1FP1 level
+      - 1 # MODE_X2_TI2  Counter counts up/down on TI1FP1 edge depending on TI2FP2 level
+      - 2 # MODE_X4_TI12 Counter counts up/down on both TI1FP1 and TI2FP2 edges
```

Comment:
Please move details into description, otherwise  they won't be displayed in doc.
See https://docs.zephyrproject.org/latest/build/dts/api/bindings/sensor/st%2Cstm32-qdec.html#dtbinding-st-stm32-qdec